projects/plugins/super-cache/wp-cache.php

```diff
@@ -3207,113 +3207,131 @@ function wp_cron_preload_cache() {
 		$wp_cache_preload_email_me = 0;
 		wp_cache_setting( 'wp_cache_preload_email_me', 0 );
 	}
-	if (
-		$wp_cache_preload_email_me &&
-		$c === 0 &&
-		! get_transient( 'taxonomy_preload' )
-	) {
-		wp_mail( get_option( 'admin_email' ), sprintf( __( '[%1$s] Cache Preload Started', 'wp-super-cache' ), home_url(), '' ), ' ' );
-	}
+
+	$just_started_preloading = false;
 
 	/*
 	 * Preload taxonomies first.
 	 *
-	 * The transient variable "taxonomy_preload" is used as a flag to indicate
-	 * that preloading of taxonomies is ongoing. It is deleted when the preload
-	 * is complete or the site owner cancels the preload.
-	 * If this flag is not set, and if a file containing a list of taxonomies
-	 * exists from a previous preload exists, it will be deleted and a new list
-	 * generated for a brand new preload.
 	 */
-	if ( $wp_cache_preload_posts == 'all' || $c < $wp_cache_preload_posts ) {
+	if ( isset( $wp_cache_preload_taxonomies ) && $wp_cache_preload_taxonomies ) {
 		wp_cache_debug( 'wp_cron_preload_cache: doing taxonomy preload.', 5 );
 		$permalink_counter_msg = $cache_path . "preload_permalink.txt";
-		if ( isset( $wp_cache_preload_taxonomies ) && $wp_cache_preload_taxonomies ) {
-			$taxonomies = apply_filters( 'wp_cache_preload_taxonomies', array( 'post_tag' => 'tag', 'category' => 'category' ) );
+		$taxonomies            = apply_filters(
+			'wp_cache_preload_taxonomies',
+			array(
+				'post_tag' => 'tag',
+				'category' => 'category',
+			)
+		);
 
-			$preload_more_taxonomies = false;
+		$preload_more_taxonomies = false;
 
-			foreach( $taxonomies as $taxonomy => $path ) {
-				$taxonomy_filename = $cache_path . "taxonomy_" . $taxonomy . ".txt";
+		foreach ( $taxonomies as $taxonomy => $path ) {
+			$taxonomy_filename = $cache_path . 'taxonomy_' . $taxonomy . '.txt';
 
-				if ( ! get_transient( 'taxonomy_preload' ) ) {
-					@unlink( $taxonomy_filename );
-				}
-				set_transient( 'taxonomy_preload', 1, 600 );
+			// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+			if ( false === @file_exists( $taxonomy_filename ) ) {
 
-				if ( false == @file_exists( $taxonomy_filename ) ) {
-					$out = '';
-					$records = get_terms( $taxonomy );
-					foreach( $records as $term ) {
-						$out .= get_term_link( $term ). "\n";
-					}
-					$fp = fopen( $taxonomy_filename, 'w' );
-					if ( $fp ) {
-						fwrite( $fp, $out );
-						fclose( $fp );
-					}
-					$details = explode( "\n", $out );
-				} else {
-					$details = explode( "\n", file_get_contents( $taxonomy_filename ) );
+				if ( ! $just_started_preloading && $wp_cache_preload_email_me ) {
+					// translators: 1: site url, 2: blank space
+					wp_mail( get_option( 'admin_email' ), sprintf( __( '[%1$s] Cache Preload Started', 'wp-super-cache' ), home_url(), '' ), ' ' );
```

Comment:
Why is there a blank space needed here?

Reply:
I found that wp_mail() refused to send a blank email in the past. I'm not sure if that has changed since, however.

Reply:
Today I learned - thanks! I didn't know you can't use `wp_mail` with an empty message. I checked the code and it's intended. Sending an empty string results in a `WP_Error`:
```
Message body empty
```

However, I was referring to the blank space in the translators line [here](https://github.com/Automattic/jetpack/pull/31022/files/bfd32c460a106b90d7ea24e1e9af364cd7d729af#diff-49e24876fd767aa7d12d7018a180f19e42dc62b47484a137fe4d42fdb31d10f6R3237). Only the message is translated, the blank space isn't.

Reply:
oh, thanks! Will fix that.